doc/nrf/app_dev/device_guides/nrf54l/nrf54l_signing_with_payload.rst

```diff
@@ -0,0 +1,238 @@
+            source-memory = <&cpuflpr_code_partition>;
+      };
+
+   where
+
+   * The ``execution-memory`` links SRAM definitions to FLPR (``cpuflpr_sram_code_data``).
+   * The ``source-memory`` links RRAM definitions to FLPR (``cpuflpr_code_partition``).
+
+   These links inform the main application about the memory usage by the FLPR core.
+
+Partition Manager configuration
+-------------------------------
+
+Adjust the static definitions in the Partition Manager as follows:
+
+* For the FLPR partition (``flpr0``):
+
+   .. code-block:: dts
+
+      flpr0:
+        address: 0x98000
+        end_address: 0xb8000
+        region: flash_primary
+        size: 0x20000
+
+  The ``address`` and ``size`` must match the DTS definitions.
+  The ``end_address`` is the sum of the address and size.
+
+* For the ``mcuboot_primary_app`` configuration:
+
+   .. code-block:: dts
+
+      mcuboot_primary_app:
+        address: 0xc800
+        end_address: 0xb8000
+        orig_span: &id002
+        - app
+        - flpr0
+        region: flash_primary
+        size: 0xab800
+        span: *id002
+
+  This configuration indicates that ``flpr0`` is now set within ``mcuboot_primary_app``, meaning it is part of the image.
+
+Building project
+****************
+
+When working with your project, you must build the CPUAPP application separately from the FLPR application.
+For detailed instructions on how to build, see :ref:`building_nrf54l`.
+
+Creating a single image
+***********************
+
+After the build is complete, you must manually collect the artifacts into a single image for MCUboot to work correctly.
+
+Merging binaries
+================
+
+Before merging, ensure that the CPUAPP and FLPR applications have been built separately and using sysbuild.
+
+1. Locate the :file:`zephyr.hex` files for both applications:
+
+   * For CPUAPP, find the :file:`build/<cpuapp_application_dir_name>/zephyr/zephyr.hex` file.
+   * For FLPR, find the :file:`<flpr_build>/zephyr/zephyr.hex` file.
+
+#. If you do not have your environment set up, first source the :file:`zephyr_env.sh` script or ensure you have set the ``ZEPHYR_BASE`` environmental variable to the Zephyr directory used for building.
+
+#. Execute the merge command, by running the following script:
+
+   .. code-block:: console
+
+      python3 ${ZEPHYR_BASE}/scripts/build/mergehex.py <flpr_build>/zephyr/zephyr.hex build/<cpuapp_application>/zephyr/zephyr.hex -o app_and_flpr_merged.hex
+
+   It results in creating the :file:`app_and_flpr_merged.hex` file that contains both the FLPR and CPUAPP application.
+
+   .. note::
+
+      In case you encounter any merging errors indicating that memory locations are defined in both files, it means that the DTS and Partition Manager partition definitions are not aligned, and the FLPR has been built in an area already occupied by the application. To resolve this issue, you must check and adjust the settings in either the Partition Manager, the DTS, or both.
```

Comment:
```suggestion
      Merging errors indicating that memory locations are defined in both hex files mean that the DTS and Partition Manager partition definitions, for FLPR and CPUAPP, are not properly aligned and the FLPR has been built to take an area already occupied by the application. To resolve this issue, you must check and adjust the settings in either the Partition Manager, the DTS, or both,  so that FLPR designated are does not overflow into application designated area.
```